.github/workflows/test.yml

```diff
@@ -0,0 +1,90 @@
+name: Test
+
+on: push
+
+env:
+  DOCKER_HUB_USERNAME: ${{ github.repository_owner }}
+  DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+  COMPOSE_FILE: docker-compose.test.yml
+  DOCKER_IMAGE: ${{ github.repository }}
+
+jobs:
+  build:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: ./.github/actions/branch-tag
```

Comment:
IS this really needed? Why not using `github.head_ref` ([official docs](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions)) in `env`:

![image](https://user-images.githubusercontent.com/696529/88256525-bd13f400-cce5-11ea-8304-ca98a9452f2d.png)

```
env:
  ...
  DOCKER_IMAGE: ${{ github.repository }}
  BRANCH_TAG: ${{ github. head_ref }}
```

I am not just not clear why this is needed 😓

Reply:
Because it is only available for `pull_request` event, but I'm using `push`, that why I need to create an action to get `BRANCH_TAG`.
But if I use `BRANCH_TAG: ${{ github. head_ref }}`, some branch names that have `/` (`chore/setup`) will be invalid for naming docker tag. In semaphore CI, we also replace `/` by `-` for `BRANCH_TAG`